src/com/hotmail/kalebmarc/textfighter/farm/Field.java

```diff
@@ -0,0 +1,130 @@
+package com.hotmail.kalebmarc.textfighter.farm;
+
+import com.hotmail.kalebmarc.textfighter.main.Random;
+import com.hotmail.kalebmarc.textfighter.main.Ui;
+
+import java.util.ArrayList;
+
+public class Field {
+    private Seed seed;
+    private int remainingTime;
+
+    public Seed getSeed() {
+        return seed;
+    }
+
+    public void menu(Farm farm, int fieldNr) {
+        while (true) {
+            // int i = 1;
+            Ui.cls();
+            Ui.printhr();
+            Ui.println(Ui.getCentred("Field " + fieldNr + " (" + farm.getName() + ") "));
+            Ui.print("(" + ((seed == null) ? "No seeds planted)" : seed.getName() + ") - "));
+            Ui.println((seed == null) ? "" : ((this.getRemainingTime() > 0) ? this.getRemainingTime() + " fights remaining until the harvest." : "Ready for harvest"));
+            Ui.println();
+// Can you think of any way I could possibly make this work? I wanted to make the menu dynamic - display the harvest option only if the crops are ready for harvest - but I just can't come up with a universal and scalable way to solve this other than bunch of if-else statements, which is rather bulky and doesn't meet either specified requirement.
```

Comment:
Here is the first comment.

Reply:
Unfortunately I couldn't think of a clean way of doing this without getting into some much more advanced stuff (a seperate helper class that can take an array of functions/items, and dynamically display them in order without any hard-coded number). I think a few if statements is good enough in this case. This is what I got (similar to yours): ```    boolean readyToHarvest = false;

    print("1) Plant Seeds");
    if (readyToHarvest) print("2) Harvest");
    print(readyToHarvest ? "3" : "2" + ") Back");

    switch (getValidInt()) {
      case 1:
        plant();
        return;
      case 2:
        if (readyToHarvest) {
          harvest();
          return;
        } else {
          print("quit!");
          return;
        }
      case 3:
        if (readyToHarvest) {
          print("quit!");
          return;
        }
      default:
        print("fallback");
    }```

(Done in a sandbox test environment, would need tweaking to work in this codebase)

----

For the situation where there are an unknown amount of extra options, theres an example of how this could be done here: https://github.com/hhaslam11/Text-Fighter/blob/959e3a2695e369243d3907a0dd8052c7111ee681/src/com/hotmail/kalebmarc/textfighter/main/Shop.java#L136

although its not really a clean solution either.

If there was to be some nice automated menu-builder classs, that should be done on a seperate PR/issue since it would be a pretty big project & would include converting all the old menus to use the new helper class